src/fondue/Fondue.js

```diff
@@ -277,6 +277,29 @@ export default class Fondue {
 		return tables.filter((table) => colorTables.includes(table));
 	}
 
+	get colorPalettes() {
+		let palettes = [];
+
+		// Create padded hex value for color string
+		const hex = (d) => Number(d).toString(16).padStart(2, "0");
+
+		// Convert color records to RGBA hex strings
+		// TODO: how are multiple palettes handled???
+		const cpal = this._font.opentype.tables.CPAL;
+		if (cpal) {
+			cpal.colorRecords.map((clr) => {
```

Comment:
Since you're mapping already, you might as well `return` immediately.

```
if(cpal) {
  return cpal.colorRecords.map(...)
}
```

Reply:
The current code makes clear that an empty `palettes` array will be returned, if there is no `CPAL` in the font. (Or in other words, it will be filled when `CPAL` is present). I personally find that a bit more explicit than having this "trailing return":

```javascript
	get colorPalettes() {
		// Create padded hex value for color string
		const hex = (d) => Number(d).toString(16).padStart(2, "0");

		// Convert color records to RGBA hex strings
		// TODO: how are multiple palettes handled???
		const cpal = this._font.opentype.tables.CPAL;
		if (cpal) {
			return cpal.colorRecords.map((clr) => {
				palettes.push(
					`#${hex(clr.red)}` +
						`${hex(clr.green)}` +
						`${hex(clr.blue)}` +
						`${hex(clr.alpha)}`
				);
			});
		}

		return [];
	}
```

Mind if I keep it as-is?

Reply:
I don't think it makes that clear? The empty variable is declared at the top of the function and you really have to read the whole function to find out it remains empty if there's no CPAL table in the font.

What do you think of the following? This uses a guard statement to make it immediately clear that an empty array will be returned if there's no CPAL table.

```js
get colorPalettes() {
  const cpal = this._font.opentype.tables.CPAL;
  if(! cpal) return [];

  // Create padded hex value for color string
  const hex = (d) => Number(d).toString(16).padStart(2, "0");
  
  return cpal.colorRecords.map((clr) => {
		palettes.push(
			`#${hex(clr.red)}` +
				`${hex(clr.green)}` +
				`${hex(clr.blue)}` +
				`${hex(clr.alpha)}`
		);
	});
}
```